core/domain/exp_domain_test.py

```diff
@@ -3516,5 +3616,60 @@ def test_that_mapping_is_correct_for_series_of_changes(self):
         }
         self.assertEqual(
             new_mapping.exploration_version, new_exploration.version)
-        self.assertEqual(new_mapping.state_names_to_ids, expected_mapping)
+        self.assertDictEqual(new_mapping.state_names_to_ids, expected_mapping)
         self.assertEqual(new_mapping.largest_state_id_used, 5)
+
+    def test_for_explorations_having_old_states_schema_version(self):
+        """Test that correct state id mapping is generated for explorations
+        having old states schema version.
+        """
+
+        # Make sure that END is present in generated state id mapping if
+        # exploration contains rules which have END as their destination state
+        # but exploration itself does not have END state.
+        old_exploration = exp_domain.Exploration.from_untitled_yaml(
+            self.EXP_ID, 'Title', 'Category', self.EXPLORATION_CONTENT_1)
+        expected_mapping = {
+            '(untitled state)': 0,
+            'New state': 1,
+            'END': 2
+        }
+        state_id_map = (
+            exp_domain.StateIdMapping.create_mapping_for_new_exploration(
+                old_exploration))
+        self.assertDictEqual(state_id_map.state_names_to_ids, expected_mapping)
+        self.assertEqual(state_id_map.largest_state_id_used, 2)
+
+
```

Comment:
To bring this back to your attention: I was actually asking the extra newline be removed since this is separating tests not test classes.